### DISCUSSION TITLE
Store XYZ in an array onClick

### DISCUSSION THREAD

**Discussion by wirmachenbunt** [cables-gl/cables_docs#280] [Answers & Questions] [answered]
i'm trying to save XYZ on mouseclick into an array. Before making my own custom op for this, maybe there is something out of the box ?

look at this patch, this isn't working since ArrayBuffer3 doesn't work when not connected to the mainloop trigger

<img width="884" alt="image" src="https://user-images.githubusercontent.com/1138673/174797585-217652bb-b1db-457d-8dd3-f63625b57919.png">

edit: ArrayBuffer3 actually does work like this, it saves the coordinates but it never reaches the simplespline

**Comment by steam0r** [comment]
wouldnt variables help with this? or maybe any of the freeze-ops if it needs to survive a reload? if you want to store only on click check vartriggernumber

**Reply by wirmachenbunt** [reply]
i want to collect xyz positions from a mouse to a growing array
doing it in a custom op right now.

**Comment by pandrr** [comment, marked as answer]
https://cables.gl/op/Ops.Array.Array3PushNumbers_v2
?

**Comment by wirmachenbunt** [comment]
exactly, this is it. custom op done anyway, now i can choose :)

cheers
